src/transxchange/read.rs

```diff
@@ -647,6 +665,7 @@ fn load_routes_vehicle_journeys_calendars(
             vj_id
         };
         let dates = create_calendar_dates(transxchange, vehicle_journey, max_end_date)?;
+        let dup_calendar = find_duplicate_calendar(collections, &calendars, &dates);
```

Comment:
Can you just move this line under the next one (dates.is_empty) ? Best perf.

Reply:
Good idea, let me try that.